templates/c/src/tic80.h

```diff
@@ -0,0 +1,262 @@
+// based on WASM-4 template: https://wasm4.org/docs
+// & zig tic-80 template
+
+#pragma once
+
+#include <stdint.h>
+#include <stdbool.h>
+
+#define WASM_EXPORT(name) __attribute__((export_name(name)))
+#define WASM_IMPORT(name) __attribute__((import_name(name)))
+
+// ┌───────────────────────────────────────────────────────────────────────────┐
+// │                                                                           │
+// │ Callbacks                                                                 │
+// │                                                                           │
+// └───────────────────────────────────────────────────────────────────────────┘
+
+WASM_EXPORT("TIC") void TIC ();
+WASM_EXPORT("BDR") void BDR (uint32_t row);
+WASM_EXPORT("SCN") void SCN (uint32_t row);
+WASM_EXPORT("OVR") void OVR ();
+
+// ┌───────────────────────────────────────────────────────────────────────────┐
+// │                                                                           │
+// │ Platform Constants                                                        │
+// │                                                                           │
+// └───────────────────────────────────────────────────────────────────────────┘
+
+#define WIDTH 240
+#define HEIGHT 136
+
+// ┌───────────────────────────────────────────────────────────────────────────┐
+// │                                                                           │
+// │ Memory Addresses                                                          │
+// │                                                                           │
+// └───────────────────────────────────────────────────────────────────────────┘
+
+#define FRAMEBUFFER        ((uint8_t*) 0)
+#define TILES              ((uint8_t*) 0x4000)
+#define SPRITES            ((uint8_t*) 0x6000)
+#define MAP                ((uint8_t*) 0x8000)
+#define GAMEPADS           ((uint8_t*) 0xFF80)
+#define MOUSE              ((uint8_t*) 0xFF84)
+#define KEYBOARD           ((uint8_t*) 0xFF88)
+#define SFX_STATE          ((uint8_t*) 0xFF8C)
+#define SOUND_REGISTERS    ((uint8_t*) 0xFF9C)
+#define WAVEFORMS          ((uint8_t*) 0xFFE4)
+#define SFX                ((uint8_t*) 0x100E4)
+#define MUSIC_PATTERNS     ((uint8_t*) 0x11164)
+#define MUSIC_TRACKS       ((uint8_t*) 0x13E64)
+#define SOUND_STATE        ((uint8_t*) 0x13FFC)
+#define STEREO_VOLUME      ((uint8_t*) 0x14000)
+#define PERSISTENT_RAM     ((uint8_t*) 0x14004)
+#define PERSISTENT_RAM_u32 ((uint8_t*) 0x14004)
```

Comment:
This should be typed `uint_32`, that's the point of a sep named define...

Reply:
good catch.